code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -139,50 +142,94 @@
-	target_types += /obj/effect/decal/cleanable/egg_smudge
-	target_types += /obj/effect/decal/cleanable/pie_smudge
-	target_types += /obj/effect/decal/cleanable/flour
-	target_types += /obj/effect/decal/cleanable/ash
-	target_types += /obj/effect/decal/cleanable/greenglow
-	target_types += /obj/effect/decal/cleanable/dirt
-	target_types += /obj/effect/decal/cleanable/deadcockroach
+	target_types = list(
+		/obj/effect/decal/cleanable/oil,
+		/obj/effect/decal/cleanable/vomit,
+		/obj/effect/decal/cleanable/robot_debris,
+		/obj/effect/decal/cleanable/crayon,
+		/obj/effect/decal/cleanable/molten_item,
+		/obj/effect/decal/cleanable/tomato_smudge,
+		/obj/effect/decal/cleanable/egg_smudge,
+		/obj/effect/decal/cleanable/pie_smudge,
+		/obj/effect/decal/cleanable/flour,
+		/obj/effect/decal/cleanable/ash,
+		/obj/effect/decal/cleanable/greenglow,
+		/obj/effect/decal/cleanable/dirt,
+		/obj/effect/decal/cleanable/deadcockroach
+		)
 
 	if(blood)
-		target_types += /obj/effect/decal/cleanable/xenoblood/
-		target_types += /obj/effect/decal/cleanable/xenoblood/xgibs
-		target_types += /obj/effect/decal/cleanable/blood/
-		target_types += /obj/effect/decal/cleanable/blood/gibs/
-		target_types += /obj/effect/decal/cleanable/blood/drip/
+		target_types += /obj/effect/decal/cleanable/xenoblood
+		target_types += /obj/effect/decal/cleanable/blood
 		target_types += /obj/effect/decal/cleanable/trail_holder
 
-/mob/living/simple_animal/bot/cleanbot/proc/clean(obj/effect/decal/cleanable/target)
-	anchored = 1
-	icon_state = "cleanbot-c"
-	visible_message("<span class='notice'>[src] begins to clean up [target].</span>")
-	mode = BOT_CLEANING
-	spawn(50)
-		if(mode == BOT_CLEANING)
-			qdel(target)
-			anchored = 0
-			target = null
-		mode = BOT_IDLE
-		icon_state = "cleanbot[on]"
+	if(pests)
+		target_types += /mob/living/simple_animal/cockroach
+		target_types += /mob/living/simple_animal/mouse
+
+	if(trash)
+		target_types += /obj/item/trash
+
+/mob/living/simple_animal/bot/cleanbot/UnarmedAttack(atom/A)
+	if(istype(A, /obj/effect/decal/cleanable))
+		anchored = 1
+		icon_state = "cleanbot-c"
+		visible_message("<span class='notice'>[src] begins to clean up [A].</span>")
+		mode = BOT_CLEANING
+		spawn(50)
+			if(mode == BOT_CLEANING)
+				if(A && isturf(A.loc))
+					var/atom/movable/AM = A
+					if(istype(AM, /obj/effect/decal/cleanable))
+						qdel(AM)
+
+				anchored = 0
+				target = null
+			mode = BOT_IDLE
+			icon_state = "cleanbot[on]"
+	else if(istype(A, /obj/item))
+		visible_message("<span class='danger'>[src] sprays hydrofluoric acid at [A]!</span>")
+		playsound(src.loc, 'sound/effects/spray2.ogg', 50, 1, -6)
+		A.acid_act(75, 10)
+	else if(istype(A, /mob/living/simple_animal/cockroach) || istype(A, /mob/living/simple_animal/mouse))
+		var/mob/living/simple_animal/M = target
+		if(!M.stat)
+			visible_message("<span class='danger'>[src] smashes [target] with it's mop!</span>")
+			M.death()
```

Comment:
GODMODE prevents damage and things like Stun, Weaken, etc. It doesn't stop death or gib or being qdelled.